### PR TITLE
Rework Popover to avoid warnings from react

### PR DESCRIPTION
**Pull request by uladzimirdev**
## Description

The better version of https://github.com/metabase/metabase/pull/33354

- ~Drops ReactDOM.findDOMNode as we pass only refs to the component~ - covered in https://github.com/metabase/metabase/pull/33367
- drops reactUnmountAt as it's recommended to use only with the root component of react app and it's going to be removed in v18 of react
- simplified render method

### CODE REVIEW

File: frontend/src/metabase/components/Popover/Popover.jsx [hunk 1]
```diff
@@ -201,11 +202,12 @@ export default class Popover extends Component {
       this._tether.destroy();
       delete this._tether;
     }
+
     if (this._popoverElement) {
-      ReactDOM.unmountComponentAtNode(this._popoverElement);
```
Comment (uladzimirdev) [inline frontend/src/metabase/components/Popover/Popover.jsx:205]: I'm not 100% sure about it, but we do not add any event listener manually and seems remove this element from DOM correctly

**Comment by uladzimirdev** [comment]
similar to https://github.com/metabase/metabase/pull/33367

File: frontend/src/metabase/components/Popover/Popover.jsx [hunk 2]
```diff
@@ -367,13 +372,33 @@ export default class Popover extends Component {
...
     }
+
     if (target == null) {
       target = this._popoverElement;
     }
+
     return target;
   }
 
-  _renderPopover(isOpen) {
```
Comment (uladzimirdev) [inline frontend/src/metabase/components/Popover/Popover.jsx:376]: diff is weird, but I moved code from `_renderPopover` to `render`

File: frontend/src/metabase/components/Popover/Popover.jsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/components/Popover/Popover.jsx:205]: I believe `unmountComponentAtNode` should be only used in pair with `ReactDOM.render`, but it's not called here

Comment (ranquild) [reply frontend/src/metabase/components/Popover/Popover.jsx:205]: Maybe previously there was no `createPortal` call but `ReactDOM.render` instead, and this is a leftover

**Review by ranquild — APPROVED** [review]

**Comment by ranquild** [comment]
@uladzimirdev let's not backport this PR for now

File: frontend/src/metabase/components/Popover/Popover.jsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/components/Popover/Popover.jsx:205]: nice catch!

**Pull request merged**